network-area-diagram/src/main/java/com/powsybl/nad/layout/Layout.java

```diff
@@ -25,4 +26,8 @@ public interface Layout {
     Map<String, Point> getInitialNodePositions();
 
     Set<String> getNodesWithFixedPosition();
+
+    void setTextNodesWithFixedPosition(Map<String, TextPosition> textNodesWithFixedPosition);
```

Comment:
add a short javadoc mentioning the map is indexed by voltageLevelId. Or, better, replace it with 
```suggestion
    void setTextNodeFixedPosition(String voltageLevelId, TextPosition textPosition);
```

Reply:
In fact we could go further! So that the user does not need to build the `TextPosition` record, which could then be hidden as an inner class of `AbstractLayout`.
```suggestion
    void setTextNodeFixedPosition(String voltageLevelId, Point topLeft, Point edgeConnection);
```